Leave PLUGIN_LIBTOOL_FLAGS so that we don't have to fix all the tests as well.

--- origsrc/libpeas-1.0.0/configure.ac	2011-04-04 14:34:00.000000000 -0500
+++ src/libpeas-1.0.0/configure.ac	2011-04-10 22:57:25.058965000 -0500
@@ -405,7 +405,7 @@ AC_SUBST(LT_AGE)
 # http://www.mail-archive.com/automake@gnu.org/msg10725.htm
 
 PLUGIN_LIBTOOL_FLAGS="-module -avoid-version"
 TEST_PLUGIN_LIBTOOL_FLAGS="$PLUGIN_LIBTOOL_FLAGS -rpath /nowhere"
-LOADER_LIBTOOL_FLAGS="-module -avoid-version -export-symbols-regex peas_register_types"
+LOADER_LIBTOOL_FLAGS="-module -avoid-version -no-undefined -export-symbols-regex peas_register_types"
 AC_SUBST(PLUGIN_LIBTOOL_FLAGS)
 AC_SUBST(TEST_PLUGIN_LIBTOOL_FLAGS)
 AC_SUBST(LOADER_LIBTOOL_FLAGS)
--- origsrc/libpeas-1.0.0/loaders/c/Makefile.am	2011-03-04 03:37:10.000000000 -0600
+++ src/libpeas-1.0.0/loaders/c/Makefile.am	2011-04-11 03:25:36.913596200 -0500
@@ -18,7 +18,7 @@ libcloader_la_SOURCES = \
 	peas-plugin-loader-c.h
 
 libcloader_la_LDFLAGS = $(LOADER_LIBTOOL_FLAGS) $(GCOV_LDFLAGS)
-libcloader_la_LIBADD = $(PEAS_LIBS)
+libcloader_la_LIBADD = $(top_builddir)/libpeas/libpeas-1.0.la $(PEAS_LIBS)
 
 gcov_sources = $(libcloader_la_SOURCES)
 include $(top_srcdir)/Makefile.gcov
--- origsrc/libpeas-1.0.0/loaders/python/Makefile.am	2011-03-07 12:11:13.000000000 -0600
+++ src/libpeas-1.0.0/loaders/python/Makefile.am	2011-04-11 03:25:40.844803000 -0500
@@ -25,6 +25,7 @@ libpythonloader_la_LDFLAGS = \
 	$(GCOV_LDFLAGS)
 
 libpythonloader_la_LIBADD = \
+	$(top_builddir)/libpeas/libpeas-1.0.la \
 	$(PEAS_LIBS)			\
 	$(PYTHON_LIBS)
 
--- origsrc/libpeas-1.0.0/loaders/seed/Makefile.am	2011-03-04 03:37:10.000000000 -0600
+++ src/libpeas-1.0.0/loaders/seed/Makefile.am	2011-04-11 03:25:45.259610700 -0500
@@ -19,7 +19,7 @@ libseedloader_la_SOURCES = \
 	peas-plugin-loader-seed.h
 
 libseedloader_la_LDFLAGS = $(LOADER_LIBTOOL_FLAGS) $(GCOV_LDFLAGS)
-libseedloader_la_LIBADD = $(PEAS_LIBS) $(SEED_LIBS)
+libseedloader_la_LIBADD = $(top_builddir)/libpeas/libpeas-1.0.la $(PEAS_LIBS) $(SEED_LIBS)
 
 gcov_sources = $(libseedloader_la_SOURCES)
 include $(top_srcdir)/Makefile.gcov
--- origsrc/libpeas-1.0.0/peas-demo/plugins/helloworld/Makefile.am	2010-10-02 18:16:37.000000000 -0500
+++ src/libpeas-1.0.0/peas-demo/plugins/helloworld/Makefile.am	2011-04-11 03:41:06.306080500 -0500
@@ -14,7 +14,10 @@ libhelloworld_la_SOURCES = \
 	peasdemo-hello-world-configurable.c
 
 libhelloworld_la_LDFLAGS = $(PLUGIN_LIBTOOL_FLAGS)
-libhelloworld_la_LIBADD  = $(PEAS_LIBS) $(PEAS_GTK_LIBS)
+libhelloworld_la_LIBADD  = \
+	$(top_builddir)/libpeas-gtk/libpeas-gtk-1.0.la \
+	$(top_builddir)/libpeas/libpeas-1.0.la \
+	$(PEAS_LIBS) $(PEAS_GTK_LIBS)
 
 plugin_DATA = helloworld.plugin
 
--- origsrc/libpeas-1.0.0/peas-demo/plugins/secondtime/Makefile.am	2010-10-02 18:16:37.000000000 -0500
+++ src/libpeas-1.0.0/peas-demo/plugins/secondtime/Makefile.am	2011-04-11 03:41:18.645702000 -0500
@@ -12,7 +12,10 @@ libsecondtime_la_SOURCES = \
 	second-time.c
 
 libsecondtime_la_LDFLAGS = $(PLUGIN_LIBTOOL_FLAGS)
-libsecondtime_la_LIBADD  = $(PEAS_LIBS) $(PEAS_GTK_LIBS)
+libsecondtime_la_LIBADD  = \
+	$(top_builddir)/libpeas-gtk/libpeas-gtk-1.0.la \
+	$(top_builddir)/libpeas/libpeas-1.0.la \
+	$(PEAS_LIBS) $(PEAS_GTK_LIBS)
 
 plugin_DATA = secondtime.plugin
 
--- origsrc/libpeas-1.0.0/peas-demo/plugins/valahello/Makefile.am	2011-03-06 09:58:46.000000000 -0600
+++ src/libpeas-1.0.0/peas-demo/plugins/valahello/Makefile.am	2011-04-11 03:41:29.877721600 -0500
@@ -15,7 +15,10 @@ plugin_LTLIBRARIES = libvalahello.la
 libvalahello_la_SOURCES = peasdemo-vala-hello-plugin.vala
 
 libvalahello_la_LDFLAGS = $(PLUGIN_LIBTOOL_FLAGS)
-libvalahello_la_LIBADD  = $(PEAS_LIBS) $(PEAS_GTK_LIBS)
+libvalahello_la_LIBADD  = \
+	$(top_builddir)/libpeas-gtk/libpeas-gtk-1.0.la \
+	$(top_builddir)/libpeas/libpeas-1.0.la \
+	$(PEAS_LIBS) $(PEAS_GTK_LIBS)
 
 plugin_DATA = valahello.plugin
 
